tests/test_core.py

```diff
@@ -4363,8 +4371,8 @@ class Derived : public Base {
       }
     ''', expected=['starting main\nBase\nDerived\nOK'])
 
-  @needs_dlfcn
   @with_both_exception_handling
+  @needs_dlfcn
```

Comment:
Does this order matter?

Reply:
Yes, I think needs_dlfcn is defined to assume the function doesn't take any args.  Or something like that.